### PR TITLE
ARROW-4139: [Python][Parquet] Wrap new parquet::LogicalType, cast min/max statistics based on LogicalType

### PR DESCRIPTION
**Pull request by wesm**
This changes the behavior of `Statistics.min` and `Statistics.max` to return the decoded logical value, such as a UTF-8 string, uint64 value, timestamp, etc.  I added `min_raw` and `max_raw` attributes to return the raw physical type value

Also fixes ARROW-5166 (uint64 as int64 overflow)

I found the `isAdjustedToUTC` parameter for `TIME` logical type to be weird, so I'm changing the Arrow conversion layer to always set this so we can get the TIME_MILLIS / TIME_MICROS converted type to be set in those cases.

### CODE REVIEW

**Comment by wesm** [comment]
cc @mrocklin @xhochy @tpboudreau

**Comment by wesm** [comment]
@crepererum this addresses your bug report ARROW-5166

File: cpp/src/parquet/arrow/schema.cc [hunk 1]
```diff
@@ -703,15 +703,18 @@ Status FieldToNode(const std::shared_ptr<Field>& field,
       break;
     case ArrowTypeId::TIME32:
       type = ParquetType::INT32;
-      logical_type = LogicalType::Time(false, LogicalType::TimeUnit::MILLIS);
+      logical_type =
+          LogicalType::Time(/*is_adjusted_to_utc=*/true, LogicalType::TimeUnit::MILLIS);
```
Comment (tpboudreau) [inline cpp/src/parquet/arrow/schema.cc]: This change and the two immediately below work for me.

**Comment by mrocklin** [comment]
Woot.  cc @rjzamora

**Comment by tpboudreau** [comment]
The core c++ library changes look good to me.  Don't have enough Cython to do a thorough review of those changes, but logic looks right (except of course for the umlauts).

**Comment by wesm** [comment]
+1. @xhochy if you have feedback I will be happy to address as follow up

**Comment by mrocklin** [comment]
Thank you for doing this @wesm .  I appreciate it.

**Comment by wesm** [comment]
No problem

**Pull request closed without merging**